Index: webapp/graphite/render/functions.py
===================================================================
--- webapp/graphite/render/functions.py	(revision 6365)
+++ webapp/graphite/render/functions.py	(working copy)
@@ -1074,6 +1074,12 @@
       last = safeLast(series)
       maximum = safeMax(series)
       minimum = safeMin(series)
+
+      if safeLen(series) is not None and safeSum(series) is not None:
+        avg = safeSum(series)/safeLen(series)
+      else:
+        avg = None
+
       if last is None:
         last = NAN
       else:
@@ -1087,12 +1093,16 @@
         minimum = NAN
       else:
         minimum = fmt(float(minimum))
-        
-      series.name = "%*s Current:%*s Max:%*s Min:%*s " % \
+      if avg is None:
+        avg = NAN
+      else:
+        avg = fmt(float(avg))
+
+      series.name = "%*s Cur: %s Max: %s Avg: %s" % \
           (-nameLen, series.name,
-            -lastLen, last,
-            -maxLen, maximum,
-            -minLen, minimum)
+          last,
+          maximum,
+          avg)
   return seriesList
 
 def aliasByNode(requestContext, seriesList, *nodes):
