Index: webapp/content/js/dashboard.js
===================================================================
--- webapp/content/js/dashboard.js	(revision 6367)
+++ webapp/content/js/dashboard.js	(revision 6368)
@@ -442,7 +442,7 @@
       '<div class="graph-container">',
         '<div class="graph-overlay">',
           '<img class="graph-img" src="{url}" width="{width}" height="{height}">',
-          '<div class="overlay-close-button" onclick="javascript: graphClicked({index}, event)">*</div>', // FIXME
+          '<div class="overlay-close-button" onclick="javascript: graphClicked({index}, event)">*</div>',
         '</div>',
       '</div>',
     '</tpl>',
@@ -580,12 +580,13 @@
         fn: function(box, e) {
 		  var until = decodeURIComponent(/until=([^&]*)/.exec(e.target.src)[1]);
 		  var from = decodeURIComponent(/from=([^&]*)/.exec(e.target.src)[1]);
-		  var fromHours = /^-(\d+)hours$/.exec(from);
+		  var fromHours = /^-(\d+)(minutes|hours|days|weeks|months)$/.exec(from);
+		  var fromMult = [60, 3600, 86400, 7*86400, 30*86400];
 		  if (fromHours)
-		    from = parseInt(new Date() / 1000) - parseInt(fromHours[1]) * 3600;
+		    from = parseInt(new Date() / 1000) - parseInt(fromHours[1]) * fromMult['minutes hours days weeks months'.split(' ').indexOf(fromHours[2])];
 		  else
 		    from = parseInt(from);
-		  if (until == '-')
+		  if (until == '-' || until == 'now')
 		    until = parseInt(new Date() / 1000);
 		  else
 			until = parseInt(until);
@@ -593,18 +594,59 @@
 		  var width = until - from;
 		  until = parseInt(from + width * (box.left + box.width));
 		  from = parseInt(from + width * box.left);
+		  if (!e.target.hasAttribute("origsrc"))
+		    e.target.setAttribute("origsrc", e.target.src);
 		  e.target.src = e.target.src.replace(/from=([^&]*)/, 'from=' + from).replace(/until=([^&]*)/, 'until=' + until);
 		}
       },
-	  dblclick: {
-        fn: function(graphView, graphIndex, element, evt) {
+	  click: {
+	    fn: function(graphView, graphIndex, element, evt) {
+          if (new Date() - lastDragEvent < 500)
+            return;
           var record = element.querySelector('img');
 		  if (!record) {
             return;
 		  }
 
-		  record.src = record.src.replace(/from=([^&]*)/, 'from=-12hours').replace(/until=([^&]*)/, 'until=-');
+		  if (record.getAttribute("origsrc"))
+		    record.src = record.getAttribute("origsrc");
 		}
+	  },
+	  dblclick: {
+        fn: function(graphView, graphIndex, element, evt) {
+		  if (GraphSize.width == defaultGraphParams.width)
+		  {
+            GraphSize.width = window.innerWidth - 64;
+		    GraphSize.height = window.innerHeight - 80;
+		  }
+		  else
+		  {
+            GraphSize.width = defaultGraphParams.width;
+		    GraphSize.height = defaultGraphParams.height;
+		  }
+		  /*
+		     graphStore.each(function (item, index) {
+    var params = {};
+    Ext.apply(params, defaultGraphParams);
+    Ext.apply(params, item.data.params);
+    Ext.apply(params, GraphSize);
+    params._uniq = Math.random();
+    if (params.title === undefined && params.target.length == 1) {
+      params.title = params.target[0];
+    }
+    if (!params.uniq === undefined) {
+        delete params["uniq"];
+    }
+    item.set('url', '/render?' + Ext.urlEncode(params));
+    item.set('width', GraphSize.width);
+    item.set('height', GraphSize.height);
+    item.set('index', index);
+  });
+*
+		   */
+		  refreshGraphs();
+		  //FIXME
+		}
 	  }
     },
 	plugins: [new Ext.DataView.DragSelector()]
@@ -1770,9 +1812,6 @@
 var activeMenu;
 
 function graphClicked(graphIndex, evt) {
-  if (new Date() - lastDragEvent < 2000)
-    return;
-
   Ext.get('merge').hide();
   var record = graphStore.getAt(graphIndex);
   if (!record) {
