Index: webapp/graphite/render/glyph.py
===================================================================
--- webapp/graphite/render/glyph.py	(revision 6378)
+++ webapp/graphite/render/glyph.py	(revision 6379)
@@ -301,15 +301,14 @@
   def drawLegend(self, elements, unique=False): #elements is [ (name,color,rightSide), (name,color,rightSide), ... ]
     self.encodeHeader('legend')
 
-    if unique:
-      # remove duplicate names
-      namesSeen = []
-      newElements = []
-      for e in elements:
-        if e[0] not in namesSeen:
-          namesSeen.append(e[0])
-          newElements.append(e)
-      elements = newElements
+    # remove duplicate names
+    namesSeen = []
+    newElements = []
+    for e in elements:
+      if e[0] not in namesSeen:
+        namesSeen.append(e[0])
+        newElements.append(e)
+    elements = newElements
 
     # Check if there's enough room to use two columns.
     rightSideLabels = False
